connection-limit/connection_limit_user.c

```diff
@@ -34,50 +34,23 @@ static const char *__doc__ =
         "Connection limit incoming TCP connections using XDP";
 
 static int ifindex;
-static char prev_prog_map[1024];
 
-int map_fd[MAP_COUNT];
+int conn_count_map_fd = -1, tcp_conns_map_fd = -1, conn_info_map_fd = -1;
 
-static int xdp_unlink_bpf_chain(const char *map_filename) {
-    int ret = 0;
-    int key = 0;
-    int map_fd = bpf_obj_get(map_filename);
-
-    if (map_fd > 0) {
-        ret = bpf_map_delete_elem(map_fd, &key);
-        if (ret != 0)
-            log_err("Failed to remove XDP program from the chain");
-    }
-    else
-        log_info("Failed to fetch previous XDP program in the chain");
-
-    if (remove(xdp_cl_ingress_next_prog) < 0) {
-        log_warn("Failed to remove link to next XDP program in the chain");
-    }
-    return ret;
-}
-
-static void signal_handler(int sig)
-{
-    log_info("Received signal %d", sig);
-    int i = 0;
-    xdp_unlink_bpf_chain(prev_prog_map);
-    for (i=0; i<MAP_COUNT; i++) {
-        close(map_fd[i]);
-    }
-    if (info != NULL)
-        fclose(info);
-    exit(EXIT_SUCCESS);
-}
+static int get_length(const char *str);
+static int is_ipv4_loopback(uint32_t *addr4);
+static int is_ipv6_loopback(struct in6_addr *addr6);
```

Comment:
why the inconsistent types?  i.e. why not struct in_addr in line 41?

Reply:
The purpose of this method is to confirm whether the local address is a loopback address or not.   It achieves this by extracting the current connection count from procfs. When it comes to IPv4, the local address is in uint32 format, whereas for IPv6 addresses, there are multiple uint32 values. Therefore, the data is organized into a struct and then passed to this verification method.

Reply:
That doesn't answer the question. Why the inconsistent types of uint32_t vs struct in6_addr?  Why not simply uint8_t* in both cases? or struct in_addr and struct in6_addr?